testing/src/main/scala/org/http4s/testing/ArbitraryInstances.scala

```diff
@@ -670,6 +671,9 @@ trait ArbitraryInstances {
     } yield Uri(scheme, authority, path, query, fragment)
   }
 
+  implicit val cogenUri: Cogen[Uri] =
+    Cogen[String].contramap(_.renderString)
```

Comment:
This could dig into products and sums, but I'd like to refactor that structure anyway.  I think this is legit.